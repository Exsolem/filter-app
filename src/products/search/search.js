import React from 'react'
import {FormControl, InputGroup} from "react-bootstrap";
import {withRouter} from 'react-router-dom'


const Search = ({ ...props}) => {
    const inputHandler = (e) => {
        const input = e.target.value
        props.getSearchValue(input);
        console.log(e.target.value);
        props.getSearchedProducts();

        if (e.target.value) {
            props.history.push(`${props.history.location.pathname}?search=${e.target.value} `);
        } else {
            props.history.push(`${props.history.location.pathname}`)
        }
    };

    return <InputGroup className="mb-12"
                       style={{
                           margin: '2vmin 0',
                           padding: 0
                       }}>
        <FormControl onChange={inputHandler} value={props.search}/>
    </InputGroup>
};

export default withRouter(Search)