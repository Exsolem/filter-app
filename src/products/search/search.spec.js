import React from 'react'
import {shallow} from 'enzyme'

import {Search} from './search'


describe('Search', () => {
    const mockGetSearchValue = jest.fn();
    const mockGetSearchedProducts = jest.fn();
    const mockPush = jest.fn();
    const props = {
        search: 'searchValue',
        getSearchValue: mockGetSearchValue,
        getSearchedProducts: mockGetSearchedProducts,
        history: {
            push: mockPush,
            location: {
                pathname: 'pathname'
            }
        },
    };

    describe('Search initial', () => {
        const search = shallow(<Search {...props}/>);
        it('renders properly', () => {
            expect(search.debug()).toMatchSnapshot()
        });
        it('input is rendered', () => {
            expect(search.find('FormControl')).toHaveLength(1);
        });
        it('FormControl value equal to search from props', () => {
            expect(search.find('FormControl').prop('value')).toEqual(props.search);
        })
    });
    describe('when typing into input', () => {
        const search = shallow(<Search {...props}/>);

        search.find('FormControl').simulate('change', {
            target: {
                value: 'another Value',
            },
        });
        it('on input change getSearchedProducts called', () => {
            expect(mockGetSearchedProducts).toHaveBeenCalledTimes(1)
        });
        it('on input change getSearchedProducts called', () => {
            expect(mockGetSearchedProducts).toHaveBeenCalledTimes(1)
        });
        it('on input change getSearchValue called', () => {
            expect(mockGetSearchValue).toHaveBeenCalledTimes(1)
        });
        it('on input change push called', () => {
            expect(mockPush).toHaveBeenCalledTimes(1)
        })
    })
});