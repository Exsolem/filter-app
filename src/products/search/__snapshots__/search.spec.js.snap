// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Search initial renders properly 1`] = `
"<InputGroup className=\\"mb-12\\" style={{...}}>
  <FormControl onChange={[Function: inputHandler]} value=\\"searchValue\\" />
</InputGroup>"
`;
